File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) 
    {
        int n = nums.size();
        int ele = nums[0], c = 1;

        for(int i = 1; i < n; i++)
        {
            if(nums[i] == ele) c++;
            else 
            {
                c--;
                if(c == 0)
                {
                    ele = nums[i];
                    c = 1;
                }
            }
        }

        return ele;
    }
};

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        cnt = 0
        majEle = None

        for num in nums:
            if num == majEle:
                cnt += 1
            elif cnt == 0:
                majEle = num
                cnt += 1
            else:
                cnt -= 1
            
        return majEle